src/lib/y2users/useradd_config.rb

```diff
@@ -0,0 +1,186 @@
+  # traditionally stored at /etc/default/useradd, but there can be exceptions. For example, the
+  # attribute "umask" is handled by this class because it was originally a regular useradd
+  # default value. If in doubt, check the documentation of each attribute.
+  #
+  # Attributes can be nil to indicate the value is unknown or irrelevant. That can happen if the
+  # UseraddConfig object is constructed from an AutoYaST profile.
+  #
+  # When creating users in the target system, the configuration is written into that system in
+  # advance to make it available for useradd. Note that only the attributes listed at
+  # {.writable_attributes} will be actually persisted to the target system at that point. That is,
+  # those are the only default values that can really be changed in the corresponding UseraddConfig
+  # object to affect how the users are created and, thus, are the only ones with defined setters.
+  #
+  # As a software design decision, attributes that correspond to a key at /etc/default/useradd are
+  # writable only if they can be changed via "adduser -D". For example, "skel" is not writable
+  # because the support to adjust its value via "adduser -D" was removed at some point during the
+  # useradd lifecycle (it was possible to adjust it in version 3.x but not in 4.x).
+  #
+  # Many of the attributes in this class have their counterpart in the <user_defaults> section of
+  # the AutoYaST profile. But note there is not a 1:1 relationship because some of the attributes
+  # from the profile simply don't have an equivalent setter in this class, so (Auto)YaST cannot
+  # change the default value to affect the useradd behavior (eg. "skel").
+  #
+  # Moreover, there are two attributes from that AutoYaST section that don't even have a counterpart
+  # in the useradd configuration: "groups" and "no_groups". The corresponding key GROUPS was dropped
+  # from the useradd configuration with no substitute. Even if the GROUPS key is present in
+  # /etc/default/useradd, its value will be completely ignored by useradd and by YaST.
+  #
+  # The "groups", "no_groups" and "skel" attributes from the profile may still be honored by
+  # AutoYaST when creating users by any other mechanism other than UseraddConfig.
+  class UseraddConfig
+    # @see .writable_attributes
+    WRITABLE_ATTRIBUTES = [:group, :home, :umask, :expiration, :inactivity_period, :shell].freeze
+    private_constant :WRITABLE_ATTRIBUTES
+
+    class << self
+      # Names of the attributes that can be persisted to the configuration of the system and thus
+      # can be used to really modify the default useradd behavior
+      #
+      # Only attributes in this list have a public setter
+      #
+      # @return [Array<Symbol>]
+      def writable_attributes
+        WRITABLE_ATTRIBUTES.dup
+      end
+
+    private
+
+      # Internal class method to automatically define the setters for all writable attributes
```

Comment:
I worry this way documentation will miss info that ttrs are writable and setter exists at all.

Reply:
Maybe. Let me lookup whether there is a solution. So far, that very well resolved in Crystal with macros :-)

Reply:
I changed the strategy a bit and now it's properly documented in the yardoc output.